src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -394,26 +392,37 @@ public bool IsSolutionAvailable
                 var projectPaths = new List<string>();
                 IEnumHierarchies enumHierarchies;
                 var guid = Guid.Empty;
-                var hr = _vsSolution.GetProjectEnum((uint)__VSENUMPROJFLAGS3.EPF_DEFERRED, ref guid, out enumHierarchies);
 
-                ErrorHandler.ThrowOnFailure(hr);
+                var vsSolution = _serviceProvider.GetService<SVsSolution, IVsSolution>();
 
-                // Loop all projects found
-                if (enumHierarchies != null)
+                if (vsSolution != null)
                 {
-                    // Loop projects found
-                    var hierarchy = new IVsHierarchy[1];
-                    uint fetched = 0;
-                    while (enumHierarchies.Next(1, hierarchy, out fetched) == VSConstants.S_OK && fetched == 1)
-                    {
-                        string projectPath;
-                        hierarchy[0].GetCanonicalName(VSConstants.VSITEMID_ROOT, out projectPath);
+                    var hr = vsSolution.GetProjectEnum((uint)__VSENUMPROJFLAGS3.EPF_DEFERRED, ref guid, out enumHierarchies);
 
-                        if (!string.IsNullOrEmpty(projectPath))
+                    ErrorHandler.ThrowOnFailure(hr);
+
+                    // Loop all projects found
+                    if (enumHierarchies != null)
+                    {
+                        // Loop projects found
+                        var hierarchy = new IVsHierarchy[1];
+                        uint fetched = 0;
+                        while (enumHierarchies.Next(1, hierarchy, out fetched) == VSConstants.S_OK && fetched == 1)
                         {
-                            projectPaths.Add(projectPath);
+                            string projectPath;
+                            hierarchy[0].GetCanonicalName(VSConstants.VSITEMID_ROOT, out projectPath);
+
+                            if (!string.IsNullOrEmpty(projectPath))
+                            {
+                                projectPaths.Add(projectPath);
+                            }
                         }
                     }
+
+                }
+                else
+                {
+                    Debug.Fail("Failed to get IVsSolution from service provider.");
```

Comment:
Debug assert should be placed right after the line where the error has occurred. Otherwise it's hard to correlate this failure with actual location. 

Instead I suggest adding this snippet at line 397
```cs
Debug.Assert(vsSolution != 0, "Failed to get IVsSolution from service provider.");
```